couchbase-entity-api/src/main/java/kaufland/com/coachbasebinderapi/scheme/DefaultSchemeValidator.kt

```diff
@@ -0,0 +1,57 @@
+package kaufland.com.coachbasebinderapi.scheme
+
+open class DefaultSchemeValidator : SchemeValidator {
```

Comment:
Nitpick: Probably "scheme" should read "schema". A scheme is something else. Maybe change now before it becomes a breaking change later.

Reply:
> > Nothing much to add to what Marco said. I'd only rename scheme to schema.
> > The Coachbase vs Couchbase is also a question of now or never I guess. The pain of changing it will only increase with time.
> 
> From my perspective when we think about doing this breaking change we should think about a whole new name since this framework have 0 hard dependencies to Couchbase instead it's usable by anything which represents data in maps.
> 
> What do you guys think?

Yes thats a good idea! Would you replace "Couchbase" by "Database"? Or "DatabaseEngine" (I am thinking along typical ORM names). What the entity framework does is a little similar to traditional ORM, right?

Reply:
yes you're right but i vote for a unconventional name like "Bam!Tity"

Reply:
BamSyntaxErrorTity more like :D